src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -82,12 +82,32 @@ case class DepositIngestTask(deposit: Deposit,
       _ = debug(s"Assigning curator role to ${deposit.depositorUserId}")
       _ <- instance.dataset(persistentId).assignRole(RoleAssignment(s"@${deposit.depositorUserId}", DefaultRole.curator.toString))
       _ <- instance.dataset(persistentId).awaitUnlock()
-      _ <- if (publish) publishDataset(persistentId)
+      publicationDate <- getPublicationdate(optAmd)
+      _ <- if (publish) publishDataset(persistentId, publicationDate)
            else keepOnDraft()
     } yield ()
     // TODO: delete draft if something went wrong
   }
 
+  def getPublicationdate(optAmd: Option[Node]): Try[String] = Try {
```

Comment:
See: https://github.com/DANS-KNAW/dd-dans-deposit-to-dataverse/blob/e92fbccd9737b8aa7c456feddad74dfcc731c60a/src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Amd.scala#L22

Reply:
Done